test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -2213,4 +2220,31 @@ private static boolean isMlEnabled() {
         }
     }
 
+    protected Map<NodeFeature, Version> getHistoricalFeatures() {
+        if (historicalFeatures == null) {
+            Map<NodeFeature, Version> historicalFeaturesMap = new HashMap<>();
+            String metadataPath = System.getProperty("tests.features.metadata.path");
+            if (metadataPath == null) {
+                throw new UnsupportedOperationException("Historical features information is unavailable when using legacy test plugins.");
+            }
+
+            String[] metadataFiles = metadataPath.split(File.pathSeparator);
```

Comment:
Is this going to give us problems on Windows? I know we've had issues with large command line arguments before, and this could be potentially large (a file per module/plugin?). Wondering if we should collect into a directory?

Reply:
This is a system property, and for Gradle test workers all command line arguments are actually read from the worker via stdin so we should be good here.

Reply:
Also, we use this same pattern for passing the locations of plugins/modules for test clusters to install and we haven't hit any issues there yet that I'm aware of.